dags/oup/repository.py

```diff
@@ -0,0 +1,60 @@
+import os
+from collections import defaultdict
+from io import BytesIO
+from typing import IO
+
+from common.repository import IRepository
+from common.s3_service import S3Service
+
+
+class OUPRepository(IRepository):
+    ZIPED_DIR: str = "raw/"
+    EXTRACTED_DIR: str = "extracted/"
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.s3 = S3Service(os.getenv("OUP_BUCKET_NAME", "oup"))
+
+    def get_all_raw_filenames(self):
+        return [
+            os.path.basename(f.key)
+            for f in self.s3.objects.filter(Prefix=self.ZIPED_DIR).all()
+        ]
+
+    def find_all(self, filenames_to_process=None):
+        ret_dict = {}
+        filenames = []
+        filenames = (
+            filenames_to_process
+            if filenames_to_process
+            else self.__find_all_extracted_files()
+        )
+        for file in filenames:
+            last_part = os.path.basename(file)
+            filename_without_extension = last_part.split(".")[0]
+            ret_dict.setdefault(filename_without_extension, defaultdict(dict))
+            extension = "xml" if self.is_meta(last_part) else "pdf"
+            ret_dict[filename_without_extension].setdefault(extension, file)
+        return list(ret_dict.values())
+
+    def find_by_id(self, id: str):
```

Comment:
maybe `get_by_id` since you are downloading it?

Reply:
makes sense, this change will cause 18 files change :D